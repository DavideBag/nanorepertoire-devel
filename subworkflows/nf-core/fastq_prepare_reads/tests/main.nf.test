// nf-core subworkflows test fastq_prepare_reads
nextflow_workflow {

    name "Test Subworkflow FASTQ_PREPARE_READS"
    script "../main.nf"
    workflow "FASTQ_PREPARE_READS"
    config "./nextflow.config"
    tag "subworkflows"
    tag "subworkflows_nfcore"
    tag "subworkflows/fastq_prepare_reads"
    tag "cutadapt"
    tag "flash"
    tag "rename"
    tag "nanotranslate"
    tag "fastqc"


    test("preprocessing - fastqc - both_end") {

        when {
            workflow {
                """
                // input_ch
                input[0] = Channel
                            .fromfilePairs("../../TESTDATA/test01_raw_R*.fastq", checkIfExists: true)
                            .map{ name, reads ->
                                [[id:'test'], [reads[0], reads[1]]]}
                input[1] = Channel.fromPath("https://github.com/lescailab/testdata/blob/master/nanobodies/adapters.fa", checkIfExists: true)
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match()}
                //TODO nf-core: Add all required assertions to verify the test output.
            )
        }
    }
}
