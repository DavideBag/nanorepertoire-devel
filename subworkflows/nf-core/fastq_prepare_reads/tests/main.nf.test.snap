{
    "preprocessing - fastq - both_end": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "4": [
                    
                ],
                "5": [
                    
                ],
                "6": [
                    
                ],
                "fastqc": [
                    
                ],
                "merged": [
                    
                ],
                "renamed": [
                    
                ],
                "translated": [
                    
                ],
                "translog": [
                    
                ],
                "trimmed_fastq": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-07-05T16:17:49.548526"
    },
    "preprocessing - fastqc - both_end": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "2": [
                    
                ],
                "3": [
                    
                ],
                "4": [
                    
                ],
                "5": [
                    
                ],
                "6": [
                    
                ],
                "fastqc": [
                    
                ],
                "merged": [
                    
                ],
                "renamed": [
                    
                ],
                "translated": [
                    
                ],
                "translog": [
                    
                ],
                "trimmed_fastq": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-07-08T11:00:35.961037"
    }
}